planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/utils.cpp

```diff
@@ -559,8 +559,8 @@ bool isEgoWithinOriginalLane(
   const lanelet::ConstLanelets & current_lanes, const Pose & current_pose,
   const BehaviorPathPlannerParameters & common_param, const double outer_margin)
 {
-  const auto lane_length = lanelet::utils::getLaneletLength2d(current_lanes);
-  const auto lane_poly = lanelet::utils::getPolygonFromArcLength(current_lanes, 0, lane_length);
+  const auto combined_lane = lanelet::utils::combineLaneletsShape(current_lanes);
```

Comment:
using `common_data_ptr->lanes_polygon_ptr->current` will allow us to save some time and skip this part. because creating polygon is a bit expensive

Reply:
@zulfaqar-azmi-t4 this function is in bpp common utils not LC utils

Reply:
Yes, can always make another overloading function, i.e, one with similar arguments, and another takes polygon2d as input.